### PR TITLE
ADS: Typography

### PR DESCRIPTION
**Pull request by malmstein**
Task/Issue URL: https://app.asana.com/0/1174433894299346/1202861107912488

### Description
This PR adds a new DaxTextView that allows for customization.

### Steps to test this PR
No TextView has been replaced in this PR, it will happen later.

### CODE REVIEW

File: common-ui/src/main/res/values/themes.xml [hunk 1]
```diff
@@ -57,6 +57,33 @@
...
+        <item name="textAppearanceH2">@style/Typography.DuckDuckGo.H2</item>
+        <item name="textAppearanceH3">@style/Typography.DuckDuckGo.H3</item>
+        <item name="textAppearanceH4">@style/Typography.DuckDuckGo.H4</item>
+        <item name="textAppearanceH5">@style/Typography.DuckDuckGo.H5</item>
+
+        <!--Design System Legacy Typography (to be removed when the migration happens)
+        <item name="textAppearanceBody1">@style/Typography.DuckDuckGo.Body1</item>
+        <item name="textAppearanceBody2">@style/Typography.DuckDuckGo.Body2</item>
+        <item name="textAppearanceButton">@style/Typography.DuckDuckGo.Button</item>
+        <item name="textAppearanceCaption">@style/Typography.DuckDuckGo.Caption</item> -->
```
Comment (aitorvs) [inline common-ui/src/main/res/values/themes.xml:85]: Should this be removed (?)

File: common-ui/src/main/java/com/duckduckgo/mobile/android/ui/view/text/DaxTextView.kt [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+                    5 -> H5
+                    6 -> Body1
+                    7 -> Body2
+                    8 -> Button
+                    9 -> Caption
+                    else -> Body1
+                }
+            }
+
+            fun getTextAppearanceStyle(type: Type): Int {
```
Comment (aitorvs) [inline common-ui/src/main/java/com/duckduckgo/mobile/android/ui/view/text/DaxTextView.kt:87]: This method should probably be `private`

File: common-ui/src/main/java/com/duckduckgo/mobile/android/ui/view/text/DaxTextView.kt [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+        H3,
+        H4,
+        H5,
+        Body1,
+        Body2,
+        Button,
+        Caption;
+
+        companion object {
+            fun from(type: Int): Type {
```
Comment (aitorvs) [inline common-ui/src/main/java/com/duckduckgo/mobile/android/ui/view/text/DaxTextView.kt:70]: This method should probably be `private`

**Review by nalcalag — APPROVED** [review]
Approved 💯

**Pull request merged**